test/AccessLock.t.sol

```diff
@@ -0,0 +1,726 @@
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams(-120, 120, 1 * 10 ** 18),
+            abi.encode(amount, AccessLockHook.LockAction.ModifyPosition)
+        );
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        // Should have less balance in both currencies.
+        assertLt(balanceOfAfter0, balanceOfBefore0);
+        assertLt(balanceOfAfter1, balanceOfBefore1);
+    }
+
+    function test_beforeModifyPosition_donate_succeedsWithAccessLock(uint128 amount) public {
+        vm.assume(amount != 0 && amount > 10 && amount < uint128(type(int128).max)); // precision
+        // Add liquidity so there is a position to receive fees.
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({tickLower: -120, tickUpper: 120, liquidityDelta: 100 * 10e18}),
+            ZERO_BYTES
+        );
+
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams(-120, 120, 1 * 10 ** 18),
+            abi.encode(amount, AccessLockHook.LockAction.Donate)
+        );
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        // Should have less balance in both currencies.
+        assertLt(balanceOfAfter0, balanceOfBefore0);
+        assertLt(balanceOfAfter1, balanceOfBefore1);
+    }
+
+    /**
+     *
+     * BEFORE SWAP TESTS
+     *
+     */
+
+    function test_beforeSwap_mint_succeedsWithAccessLock(uint128 amount) public {
+        vm.assume(amount != 0 && amount < uint128(type(int128).max));
+
+        // Add liquidity so there is something to swap against.
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({tickLower: -120, tickUpper: 120, liquidityDelta: 100 * 10e18}),
+            ZERO_BYTES
+        );
+
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        // Small amount to swap (like NoOp). This way we can expect balances to just be from the hook applied delta.
+        BalanceDelta delta = swapRouter.swap(
+            key,
+            IPoolManager.SwapParams(true, 1, TickMath.MIN_SQRT_RATIO + 1),
```

Comment:
nit: why not NoOp the swap?

Reply:
oo I can change to use NoOp now that its merged!

Reply:
Hm actually i'm going to leave as is 0 would prefer these tests don't go through the no-op code